### ISSUE TITLE
RGSoC issues

### ISSUE THREAD

**Issue by hobarrera** [pimutils/todoman#146]
Since RGSoC applicants have to chose a single task to plan during their summer of code, we're tailoring a lost of issues that applicants can choose from:

### #95: urwid testing
Write something to support testing the interactive screens (both funcitonality, and layout). IMHO, this is the most useful and interesting choice.

### #62: Full-fledged curses-based interface
This includes a list, and all cli-options accessible from a central UI. There's an incomplete PR for this, and it might be too short for RGSoC.
Maybe we can also include writing a few custom widgets that we can use:
* Pulldown widgets for list selection
* A "progress bar" like one to move urgency and completion percentage
* Probably something else.

### #17: Support for repeating reminders
Complete support for repeating reminders (using alarms, probably, this also has a bit of research), and always bumping up repeating reminders if recurrances are pending.

### #16 Checklists
Not sure if possible. @untitaker ?

**Note**: This list is not final. We've some discussion pending with @untitaker.

**Comment by untitaker** [comment]
I don't see why checklists shouldn't be "possible", though the `DESCRIPTION`-based implementation mostly benefits people who synchronize with OpenTasks.

I also consider https://github.com/pimutils/todoman/issues/144 to be a good project for rgsoc.

**Comment by hobarrera** [comment]
> I don't see why checklists shouldn't be "possible", though the DESCRIPTION-based implementation mostly benefits people who synchronize with OpenTasks.

I believe there were some limitations regarding vdirsyncer syncing?
And yes, being OpenTasks-specific, maybe we should have it as behind some flag to enable it (in settings).

> I also consider #144 to be a good project for rgsoc.

Definitely, I actually started this list because of it and forgot. Added it now.

**Comment by hobarrera** [comment]
Yeah, that doesn't sound too good. I don't think that particular task is a good candidate. Interest in it is also rather low.

**Comment by geier** [comment]
regarding #62 "Pulldown widgets for list selection": those can probably be taken from khal, there is a basic implementation by @untitaker and one which allows you to narrow the list by typing into a Text field (EditSelect in branch feature_timezones).

Perhaps it would then be time to start a new project `urwid-contrib` or something like that.

**Comment by untitaker** [comment]
todoman now has a very simple task list selector in TodoEditor.

**Comment by AnubhaAgrawal** [comment]
@hobarrera may I know is any team is selected for RGSOC 2017 from todoman.

**Comment by untitaker** [comment]
I think you should have recieved an email by rgsoc? If not please contact the organizers.

On 5 May 2017 18:31:56 GMT+02:00, Anubha Agrawal <notifications@github.com> wrote:
>@hobarrera may I know is any team is selected for RGSOC 2017 from
>todoman.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/pimutils/todoman/issues/146#issuecomment-299512585

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by hobarrera** [comment]
Hi. I got an email a few minutes ago that, regrettably, no teams have been selected for todoman.
